business/workloads.go

```diff
@@ -989,3 +991,23 @@ func controllerPriority(type1, type2 string) string {
 		return type2
 	}
 }
+
+// fillCustomDashboardRefs finds all dashboard IDs and Titles associated to this workload and add them to the model
+func (in *WorkloadService) fillCustomDashboardRefs(namespace string, workload *models.Workload) {
+	uniqueRefs := make(map[string]string)
+	for _, pod := range workload.Pods {
+		for _, ref := range pod.CustomDashboards {
+			if ref != "" {
+				uniqueRefs[ref] = ref
+			}
+		}
+	}
+	mon, err := kubernetes.NewKialiMonitoringClient()
+	if err != nil {
+		// Do not fail the whole query, just log & return
+		log.Error("Cannot initialize Kiali Monitoring Client")
+		return
+	}
+	dash := NewDashboardsService(mon, in.prom)
```

Comment:
Side comment.
Probably we want to merge all these XYZService split into one.
With the level of interdependencies we can simplify them, but of course, it has nothing to do with this PR, just to not forget.